service/history/handler.go

```diff
@@ -1580,42 +1470,120 @@ func (h *Handler) ReapplyEvents(
 		request.GetDomainUUID(),
 		execution.GetWorkflowId(),
 		execution.GetRunId(),
-		historyEvents,
+		adapter.ToProtoHistoryEvents(historyEvents),
 	); err != nil {
-		return h.error(err, scope, domainID, workflowID)
+		return nil, h.error(err, scope, domainID, workflowID)
 	}
-	return nil
+	return &historyservice.ReapplyEventsResponse{}, nil
 }
 
-// ReadDLQMessages reads replication DLQ messages
-func (h *Handler) ReadDLQMessages(
-	ctx context.Context,
-	request *r.ReadDLQMessagesRequest,
-) (resp *r.ReadDLQMessagesResponse, retError error) {
-	panic("ReadDLQMessages should not be called for thrift handler")
+func (h *Handler) ReadDLQMessages(ctx context.Context, request *historyservice.ReadDLQMessagesRequest) (_ *historyservice.ReadDLQMessagesResponse, retError error) {
+	defer log.CapturePanicGRPC(h.GetLogger(), &retError)
+
+	h.startWG.Wait()
+
+	scope := metrics.HistoryReadDLQMessagesScope
+	h.GetMetricsClient().IncCounter(scope, metrics.CadenceRequests)
+	sw := h.GetMetricsClient().StartTimer(scope, metrics.CadenceLatency)
+	defer sw.Stop()
+
+	engine, err := h.controller.getEngineForShard(int(request.GetShardID()))
+	if err != nil {
+		err = h.error(err, scope, "", "")
+		return nil, err
+	}
+
+	resp, err := engine.ReadDLQMessages(ctx, request)
+	if err != nil {
+		err = h.error(err, scope, "", "")
+		return nil, err
+	}
+
+	return resp, nil
 }
 
-// PurgeDLQMessages deletes replication DLQ messages
-func (h *Handler) PurgeDLQMessages(
-	ctx context.Context,
-	request *r.PurgeDLQMessagesRequest,
-) (retError error) {
-	panic("PurgeDLQMessages should not be called for thrift handler")
+func (h *Handler) PurgeDLQMessages(ctx context.Context, request *historyservice.PurgeDLQMessagesRequest) (_ *historyservice.PurgeDLQMessagesResponse, retError error) {
```

Comment:
I already moved this entire API to grpc.  I don't think we need the thrift handler definitions for this.

Reply:
Just making sure you completely removed the thrift handlers for history service and now only handler we register is grpc based?